working/extension_structs/overview.md

```diff
@@ -0,0 +1,815 @@
+functionality.  The second step is to support a further restriction on structs
+with a single field which eliminates the wrapper object, representing the struct
+entirely as the underlying object (at the cost of making the abstraction
+entirely static).
+
+This proposal builds on previous proposals in this space, including:
+  - [Views](https://github.com/dart-lang/language/blob/master/working/1426-extension-types/feature-specification-views.md)
+  - [Extension types](https://github.com/dart-lang/language/blob/master/working/1426-extension-types/feature-specification.md)
+  - [Protected extension types](https://github.com/dart-lang/language/issues/1467)
+
+
+This overview is not intended to be a feature specification: it is designed to
+give a brief overview of the core proposal and set out some design points for
+discussion.  If there is buy-in on these ideas, we may choose to incorporate
+them into an existing proposal, or add a new feature specification.
+
+
+## Design principles
+
+We aim to minimize the differences between structs and classes.  As much as
+possible, structs should behave as restrictions of classes, and extension
+structs should behave as further restrictions of structs.  We specifically aim
+to avoid as much as possible having different behaviors for the same concept
+(e.g. differences in scoping).
+
+We also aim to minimize the amount of new syntax required, and to maximize the
+amount of new functionality that we can provide relative to the syntactic real
+estate consumed, and the new cognitive load imposed on users.
+
+## Roadmap
+
+The first section describes the full struct feature.  The second section
+describes the restriction of the full struct to support static wrappers.
+
+## Structs
+
+This section describes how structs look and behave, largely by example.  In many
+cases, alternatives or extensions to the core proposal are described in line.
+Larger extensions are postponed to a later section.
+
+## Structs by example
+
+Structs allow you to define simple classes very compactly.  For example, we
+might model a component vector as follows:
+
+```
+struct Component(int x, int y, int z, int w = 1);
+```
+
+This short definition essentially defines a class `Component`, with four fields,
+and provides a number of convenient methods.
+
+```dart
+void test() {
+  // Call the default constructor, using the default value for w
+  var c1 = Component(0, 0, 0);
+  // Call the default constructor, explicitly passing w
+  var c2 = Component(1, 1, 1, 1);
+
+  assert(c1 == c2); // Equality is defined to be structural
+  assert(c1.hashCode == c2.hashCode); // With correspondind hashCode
+
+  print(c1.debutToString()); // Prints "Component(0, 0, 0, 1)"
+
+  print(c1.toString()); // Prints "struct"
+
+  // Generated copy method
+  var c2 = c1.copyWith(x : 2);
+  assert(c2.x == 2);
+}
+```
+
+Structs may explicitly define constructors, static members, and normal class
+members (except fields), and may implement interfaces.
+
```

Comment:
Maybe add something similar to _"Struct fields can be made private by using a `_` prefix in their name"_ here.

Reply:
Done.